README.md

```diff
@@ -1310,6 +1311,42 @@ Jwts.parser()
     // ... etc ...
 ```
 
+<a name="json-jackson-custom-parsing"></a>
+### Parsing of Custom Claim Types
+
+By default JJWT will only convert simple claim types: String, Date, Long, Integer, Short and Byte.  If you need to deserialize other types you can configure the `JacksonDeserializer` by passing a `Map` of claim names to types in through a constructor. For example:
+
+```java
+new JacksonDeserializer(Maps.of("user", User.class))
```

Comment:
Shouldn't we show the constructor with `ObjectMapper` as an argument?

Reply:
This constructor will create and manipulate a new `ObjectMapper` (conventionally mapping the claims to types).  This could cause unforeseen issues with an existing object mapper.

Anything more _advanced_ then this, the `JacksonDeserializer(ObjectMapper)` constructor should be used (as shown in the previous section).

Reply:
My point is shouldn't there be a `JacksonDeserializer(ObjectMapper, Map)` constructor?

Reply:
No, I think if you want to use a custom ObjectMapper, you would be in charge of configuring that instance.  This constructor modifies the ObjectMapper, which could invalidate, or worse only slightly change how the ObjectMapper was originally configured.

You could end up in a situation where the same ObjectMapper is used for a few requests, then a JwtParser is constructed for the first time (with `JacksonDeserializer(ObjectMapper, Map)`) and this could change how future requests were parsed with the `ObjectMapper` in that application.  For example, I couldn't use `JacksonSerializer.DEFAULT_OBJECT_MAPPER` for the same reason, as it would have changed how other JacksonDeserializers would parse.

That said this issue is specific to the readme, lets me move this over to #495 (I'll come back and tweak the doc with the signatures that we agree with)
https://github.com/jwtk/jjwt/pull/495/files#diff-43ca7dcadb15963ec66e63df02544e1aR68

Reply:
Ah, ok, this is super helpful.

I think we need to be super clear about this in the documentation. For example, the very first sentence in the [first section on Jackson](https://github.com/jwtk/jjwt#jackson-json-processor) states:

> If you have an application-wide Jackson ObjectMapper (as is typically recommended for most applications), you can eliminate the overhead of JJWT constructing its own ObjectMapper by using yours instead.

I'm sure people would be curious why they can use their own ObjectMapper in one scenario, but not another, so this should be crystal clear in the docs IMO (i.e. in this PR).